examples/tcp_client.rs

```diff
@@ -1,36 +1,155 @@
-use std::{io, net};
+#[cfg(target_os = "linux")]
+fn main() -> std::io::Result<()> {
+    use std::net;
+    use std::{io::Write, time::Duration};
 
-use polling::Event;
-use socket2::Type;
+    use polling::Event;
+    use socket2::Type;
 
-fn main() -> io::Result<()> {
+    std::thread::spawn(|| {
+        let listener = net::TcpListener::bind("0.0.0.0:8080").unwrap();
+        println!("Listening on {}", listener.local_addr().unwrap());
+        for stream in listener.incoming() {
+            let mut stream = stream.unwrap();
+            stream.write_all(b"Hello, world!\n").unwrap();
+        }
+    });
+    std::thread::sleep(Duration::from_millis(100));
     let socket = socket2::Socket::new(socket2::Domain::IPV4, Type::STREAM, None)?;
     let poller = polling::Poller::new()?;
     unsafe {
         poller.add(&socket, Event::new(0, true, true))?;
     }
-    let addr = net::SocketAddr::new(net::Ipv4Addr::LOCALHOST.into(), 8080);
+
     socket.set_nonblocking(true)?;
-    let _ = socket.connect(&addr.into());
 
     let mut events = polling::Events::new();
+    poller.wait(&mut events, Some(Duration::from_secs(3)))?;
+
+    let event = events.iter().next().expect("no event");
+
+    assert!(event.is_interrupt());
+
+    let addr = net::SocketAddr::new("127.0.0.1".parse().unwrap(), 8080);
+    let err = socket.connect(&addr.into()).unwrap_err();
+
+    // EINPROGRESS
+    assert_eq!(115, err.raw_os_error().expect("No OS error"));
 
+    poller
+        .modify(&socket, Event::writable(0))
+        .expect("modify failed");
     events.clear();
-    poller.wait(&mut events, None)?;
-
-    let event = events.iter().next();
-    let event = match event {
-        Some(event) => event,
-        None => {
-            println!("no event");
-            return Ok(());
+    poller.wait(&mut events, Some(Duration::from_secs(3)))?;
+
+    let event = events.iter().next().expect("no event");
+
+    assert!(event.writable);
+    assert!(!event.is_interrupt());
+    assert!(!event.is_err().unwrap());
+
+    println!("event: {:?}", event);
+    println!("socket is now writable");
+    // ========================================================================
+    // the below is example of a bad socket
+    println!("testing bad socket");
```

Comment:
If this is a functionality test it should be moved to be under the `tests` folder.